File: csharp/ConsultarCnpj.cs
// Consulta um CNPJ pela API da CNPJAPI (https://cnpjapi.com.br).
// Requer .NET 6+. Uso: CNPJAPI_KEY=cnpj_sua_chave dotnet run 00776574000156

using System.Net.Http.Headers;
using System.Text.Json;

string cnpj = args.Length > 0 ? args[0] : "00776574000156";
string apiKey = Environment.GetEnvironmentVariable("CNPJAPI_KEY") ?? "cnpj_sua_chave";

using var http = new HttpClient();
http.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiKey);

using var resposta = await http.GetAsync($"https://api.cnpjapi.com.br/{cnpj}");

if (!resposta.IsSuccessStatusCode)
{
    Console.Error.WriteLine($"Falha na consulta: HTTP {(int)resposta.StatusCode}");
    Environment.Exit(1);
}

using var doc = JsonDocument.Parse(await resposta.Content.ReadAsStringAsync());
var raiz = doc.RootElement;
var razao = raiz.GetProperty("RazaoSocial").GetString();
var situacao = raiz.GetProperty("SituacaoCadastral").GetProperty("Descricao").GetString();
Console.WriteLine($"{razao} - {situacao}");

File: php/consultar-cnpj.php
<?php

// Consulta um CNPJ pela API da CNPJAPI (https://cnpjapi.com.br).
// Requer PHP 8.0+ com a extensao cURL.
// Uso: CNPJAPI_KEY=cnpj_sua_chave php consultar-cnpj.php 00776574000156

$cnpj = $argv[1] ?? '00776574000156';
$apiKey = getenv('CNPJAPI_KEY') ?: 'cnpj_sua_chave';

$ch = curl_init("https://api.cnpjapi.com.br/{$cnpj}");
curl_setopt_array($ch, [
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_HTTPHEADER => ["Authorization: Bearer {$apiKey}"],
    CURLOPT_TIMEOUT => 10,
]);

$body = curl_exec($ch);
$status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

if ($status !== 200) {
    fwrite(STDERR, "Falha na consulta: HTTP {$status}\n");
    exit(1);
}

$empresa = json_decode($body, true);
echo $empresa['RazaoSocial'], ' - ', $empresa['SituacaoCadastral']['Descricao'], PHP_EOL;
